pkg/config/config.go

```diff
@@ -37,38 +37,81 @@ var (
 	validRoles = []string{"controlplane", "node"}
 )
 
-type ClusterConfig struct {
-	URL string `json:"url"`
-
-	ClusterCIDR          string `json:"clusterCIDR"`
-	ServiceCIDR          string `json:"serviceCIDR"`
-	ServiceNodePortRange string `json:"serviceNodePortRange"`
-	DNS                  string `json:"dns"`
-	Domain               string `json:"domain"`
-	MTU                  string `json:"mtu"`
+// Top level config
+type Config struct {
+	Components Components       `json:components`
+	Microshift MicroshiftConfig `json:microshift`
 }
 
-type DebugConfig struct {
-	Pprof bool `json:"pprof"`
+// Components config
+type Components struct {
```

Comment:
Did we agree on a more suitable name than "Components" (if what we end up with contains this in the serialisation at all)? For example, the "Core" that @deads2k suggested?

Reply:
`Core` sounds reasonable to me.